codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerServiceGeneratorV2.kt

```diff
@@ -42,85 +43,83 @@ class ServerServiceGeneratorV2(
     private val service = codegenContext.serviceShape
     private val serviceName = service.id.name.toPascalCase()
     private val builderName = "${serviceName}Builder"
+    private val builderPluginGenericTypeName = "Plugin"
+    private val builderBodyGenericTypeName = "Body"
+    private val builderGenerics = listOf(builderBodyGenericTypeName, builderPluginGenericTypeName).joinToString(", ")
 
     /** Calculate all `operationShape`s contained within the `ServiceShape`. */
     private val index = TopDownIndex.of(codegenContext.model)
-    private val operations = index.getContainedOperations(codegenContext.serviceShape).sortedBy { it.id }
+    private val operations = index.getContainedOperations(codegenContext.serviceShape).toSortedSet(compareBy { it.id })
 
-    /** The sequence of builder generics: `Op1`, ..., `OpN`. */
-    private val builderOps = (1..operations.size).map { "Op$it" }
-
-    /** The sequence of extension types: `Ext1`, ..., `ExtN`. */
-    private val extensionTypes = (1..operations.size).map { "Exts$it" }
-
-    /** The sequence of field names for the builder. */
-    private val builderFieldNames = operations.map { RustReservedWords.escapeIfNeeded(symbolProvider.toSymbol(it).name.toSnakeCase()) }
+    /** Associate each operation with the corresponding field names in the builder struct. */
+    private val builderFieldNames =
+        operations.associateWith { RustReservedWords.escapeIfNeeded(symbolProvider.toSymbol(it).name.toSnakeCase()) }
+            .toSortedMap(
+                compareBy { it.id },
+            )
 
-    /** The sequence of operation struct names. */
-    private val operationStructNames = operations.map { symbolProvider.toSymbol(it).name.toPascalCase() }
+    /** Associate each operation with the name of the corresponding Zero-Sized Type (ZST) struct name. */
+    private val operationStructNames = operations.associateWith { symbolProvider.toSymbol(it).name.toPascalCase() }
 
     /** A `Writable` block of "field: Type" for the builder. */
-    private val builderFields = builderFieldNames.zip(builderOps).map { (name, type) -> "$name: $type" }
+    private val builderFields =
+        builderFieldNames.values.map { name -> "$name: Option<#{SmithyHttpServer}::routing::Route<Body>>" }
+
+    /** The name of the local private module containing the functions that return the request for each operation */
+    private val requestSpecsModuleName = "request_specs"
+
+    /** Associate each operation with a function that returns its request spec. */
+    private val requestSpecMap: HashMap<OperationShape, Pair<String, Writable>> by lazy {
+        val map = HashMap<OperationShape, Pair<String, Writable>>()
+        for (operationShape in operations) {
```

Comment:
nit: this can be done functionally by mapping over `operations` into the hash map.

Reply:
Good shout - I've refactored it in https://github.com/awslabs/smithy-rs/pull/1954/commits/7452a831adaba208054812cc9146b61ac9a2e1e7